crates/nu-parser/src/flatten.rs

```diff
@@ -326,7 +327,13 @@ fn flatten_expression_into(
         }
         Expr::ExternalCall(head, args) => {
             if let Expr::String(..) | Expr::GlobPattern(..) = &head.expr {
-                output.push((head.span, FlatShape::External));
+                output.push((
+                    head.span,
+                    // If this external call is through an alias, then head.span points to the
+                    // name of the alias, but we also need the name of the aliased command. We
+                    // recover this from the span referenced by head.span_id.
+                    FlatShape::External(working_set.get_span(head.span_id)),
+                ));
```

Comment:
Gotta admit from just glancing at the diff I can't quite follow how the different span contents for the alias or the direct external result from that. :D

In the case an alias for an external has flags/args as part of this definition, where do we need to take this into account, so the resolved externals logic can ignore them appropriately?